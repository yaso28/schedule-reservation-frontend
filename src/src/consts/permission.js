export default Object.freeze({
  reservation: Object.freeze({
    read: 'reservation.read',
    write: 'reservation.write'
  })
})
