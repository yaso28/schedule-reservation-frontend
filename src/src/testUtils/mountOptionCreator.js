import { createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import routes from '@/router/routes'
import Vuex from 'vuex'
import VueI18n from 'vue-i18n'
import i18nOptions from '@/i18n/options'
import VueMeta from 'vue-meta'
import CoreuiVue from '@coreui/vue'
import icons from '@/icons'
import services from '@/services'

class MountOptionCreator {
  create (storeModules = null) {
    const localVue = createLocalVue()

    // router
    localVue.use(VueRouter)
    const router = new VueRouter({
      mode: 'history',
      base: process.env.BASE_URL,
      routes
    })

    // i18n
    localVue.use(VueI18n)
    const i18n = new VueI18n(i18nOptions)

    // vue-meta
    localVue.use(VueMeta)

    // Core UI
    localVue.use(CoreuiVue)

    const option = {
      localVue,
      router,
      i18n,
      icons,
      provide: services
    }

    // store
    if (storeModules) {
      localVue.use(Vuex)
      option.store = new Vuex.Store({
        modules: storeModules
      })
    }

    return option
  }
}

export default new MountOptionCreator()
