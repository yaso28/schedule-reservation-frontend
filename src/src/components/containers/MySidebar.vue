<template>
  <CSidebar
    :show="show"
    @update:show="showChanged"
  >
    <!-- title -->
    <CSidebarBrand>
      <span class="ml-2 font-weight-bold">{{ $t('organization_name') }}</span>
    </CSidebarBrand>

    <!-- menu begin -->
    <CSidebarNav>
      <!-- home -->
      <CSidebarNavItem
        :name="$t('home')"
        :to="{ name: 'home' }"
      />

      <!-- schedule -->
      <CSidebarNavItem
        :name="$t('schedule.model')"
        :to="{ name: 'schedule.list' }"
      />

      <!-- reservation begin -->
      <CSidebarNavDropdown
        v-if="can[permission.reservation.read]"
        :name="$t('reservation.model')"
        route="reservation"
      >
        <CSidebarNavItem
          :name="translateWords('month.model action.list')"
          :to="{ name:'reservation.month.list' }"
        />
        <CSidebarNavItem
          :name="translateWords('schedule.model action.list')"
          :to="{ name:'reservation.schedule.list' }"
        />
        <CSidebarNavItem
          :name="translateWords('reservation.model action.list')"
          :to="{ name:'reservation.reservation.list' }"
        />
        <CSidebarNavItem
          :name="$t('schedule_status.model')"
          :to="{ name:'reservation.schedule-status.list' }"
        />
        <CSidebarNavItem
          :name="$t('reservation_status.model')"
          :to="{ name:'reservation.reservation-status.list' }"
        />
        <CSidebarNavItem
          :name="$t('reservation_organization.model')"
          :to="{ name:'reservation.reservation-organization.list' }"
        />
        <CSidebarNavItem
          :name="$t('schedule_place.model')"
          :to="{ name:'reservation.schedule-place.list' }"
        />
        <CSidebarNavItem
          :name="$t('schedule_usage.model')"
          :to="{ name:'reservation.schedule-usage.list' }"
        />
        <CSidebarNavItem
          :name="$t('schedule_timetable.model')"
          :to="{ name:'reservation.schedule-timetable.list' }"
        />
        <CSidebarNavItem
          v-if="can[permission.reservation.write]"
          :name="$t('common.setting')"
          :to="{ name:'reservation.setting.edit' }"
        />
      </CSidebarNavDropdown>
      <!-- reservation end -->
    </CSidebarNav>
    <!-- menu end -->
  </CSidebar>
</template>

<script>
import { mapState, mapGetters } from 'vuex'
import permission from '@/consts/permission'

export default {
  name: 'MySidebar',
  computed: {
    ...mapState({
      show: state => state.sidebar.show
    }),
    ...mapGetters({
      isLogin: 'auth/isLogin',
      can: 'auth/can'
    }),
    permission () {
      return permission
    }
  },
  methods: {
    translateWords (name) {
      const me = this
      return name.split(' ')
        .filter(s => s.length > 0)
        .map(s => me.$t(s))
        .join('')
    },
    showChanged (value) {
      this.$store.commit('sidebar/set', value)
    }
  }
}
</script>
