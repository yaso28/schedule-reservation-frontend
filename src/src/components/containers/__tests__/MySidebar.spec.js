import { mount } from '@vue/test-utils'
import mountOptionCreator from '@/testUtils/mountOptionCreator'
import MySidebar from '../MySidebar'
import storeModules from '@/store/modules'
import permission from '@/consts/permission'

jest.mock('@/store/modules')

describe('MySidebar', () => {
  let store
  let mountOption
  let wrapper
  beforeEach(() => {
    mountOption = mountOptionCreator.create(storeModules)
    store = mountOption.store
    store.state.sidebar.show = 'responsive'
    storeModules.auth.getters.isLogin.mockReturnValue(false)
    storeModules.auth.getters.can.mockReturnValue({})
  })

  afterEach(() => {
    jest.resetAllMocks()
  })

  describe('認証なし', () => {
    beforeEach(() => {
      wrapper = mount(MySidebar, mountOption)
    })

    test('表示', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('認証あり・権限なし', () => {
    beforeEach(() => {
      storeModules.auth.getters.isLogin.mockReturnValue(true)
      wrapper = mount(MySidebar, mountOption)
    })

    test('表示', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('認証あり・権限あり', () => {
    beforeEach(() => {
      storeModules.auth.getters.isLogin.mockReturnValue(true)
    })

    describe.each([
      ['reservation.read', { [permission.reservation.read]: true }],
      ['reservation.write', { [permission.reservation.read]: true, [permission.reservation.write]: true }]
    ])('%s', (testCaseName, mockCanValue) => {
      beforeEach(() => {
        storeModules.auth.getters.can.mockReturnValue(mockCanValue)
        wrapper = mount(MySidebar, mountOption)
      })

      test('表示', () => {
        expect(wrapper.element).toMatchSnapshot()
      })
    })
  })

  describe('モバイルモード：サイドバー表示', () => {
    let backdrop
    beforeEach(() => {
      store.state.sidebar.show = true
      wrapper = mount(MySidebar, mountOption)
      backdrop = document.querySelector('div.c-sidebar-backdrop')
    })

    test('バックドロップ表示', () => {
      expect(backdrop).toBeInstanceOf(HTMLElement)
    })

    describe('バックドロップクリック', () => {
      beforeEach(() => {
        backdrop.click()
      })

      test('サイドバー隠す', () => {
        expect(storeModules.sidebar.mutations.set).toHaveBeenCalledWith(
          store.state.sidebar,
          'responsive'
        )
      })
    })
  })
})
