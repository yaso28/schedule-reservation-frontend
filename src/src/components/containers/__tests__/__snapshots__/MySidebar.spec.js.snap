// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MySidebar 認証あり・権限あり reservation.read 表示 1`] = `
<div
  class="c-sidebar c-sidebar-dark c-sidebar-lg-show c-sidebar-fixed"
>
  <div
    class="c-sidebar-brand"
  >
    <span
      class="ml-2 font-weight-bold"
    >
      やそスポーツクラブ
    </span>
  </div>
   
  <ul
    class="c-sidebar-nav h-100 ps"
    data-v-3738b722=""
    style="position: relative;"
  >
    <li
      class="c-sidebar-nav-item"
      data-v-3738b722=""
    >
      <a
        aria-current="page"
        class="router-link-exact-active c-active c-sidebar-nav-link"
        href="/"
        target="_self"
      >
        <!---->
        <!---->
         ホーム 
        <!---->
      </a>
    </li>
     
    <li
      class="c-sidebar-nav-item"
      data-v-3738b722=""
    >
      <a
        class="c-sidebar-nav-link"
        href="/schedule/list"
        target="_self"
      >
        <!---->
        <!---->
         練習予定 
        <!---->
      </a>
    </li>
     
    <li
      class="c-sidebar-nav-dropdown"
      data-v-3738b722=""
    >
      <a
        class="c-sidebar-nav-dropdown-toggle"
      >
        <!---->
        <!---->
         会場予約 
      </a>
      <ul
        class="c-sidebar-nav-dropdown-items"
      >
        <li
          class="c-sidebar-nav-item"
        >
          <a
            class="c-sidebar-nav-link"
            href="/reservation/month/list"
            target="_self"
          >
            <!---->
            <!---->
             月一覧 
            <!---->
          </a>
        </li>
         
        <li
          class="c-sidebar-nav-item"
        >
          <a
            class="c-sidebar-nav-link"
            href="/reservation/schedule/list"
            target="_self"
          >
            <!---->
            <!---->
             練習予定一覧 
            <!---->
          </a>
        </li>
         
        <li
          class="c-sidebar-nav-item"
        >
          <a
            class="c-sidebar-nav-link"
            href="/reservation/reservation/list"
            target="_self"
          >
            <!---->
            <!---->
             会場予約一覧 
            <!---->
          </a>
        </li>
         
        <li
          class="c-sidebar-nav-item"
        >
          <a
            class="c-sidebar-nav-link"
            href="/reservation/schedule-status/list"
            target="_self"
          >
            <!---->
            <!---->
             公開ステータス 
            <!---->
          </a>
        </li>
         
        <li
          class="c-sidebar-nav-item"
        >
          <a
            class="c-sidebar-nav-link"
            href="/reservation/reservation-status/list"
            target="_self"
          >
            <!---->
            <!---->
             予約ステータス 
            <!---->
          </a>
        </li>
         
        <li
          class="c-sidebar-nav-item"
        >
          <a
            class="c-sidebar-nav-link"
            href="/reservation/reservation-organization/list"
            target="_self"
          >
            <!---->
            <!---->
             予約団体 
            <!---->
          </a>
        </li>
         
        <li
          class="c-sidebar-nav-item"
        >
          <a
            class="c-sidebar-nav-link"
            href="/reservation/schedule-place/list"
            target="_self"
          >
            <!---->
            <!---->
             場所 
            <!---->
          </a>
        </li>
         
        <li
          class="c-sidebar-nav-item"
        >
          <a
            class="c-sidebar-nav-link"
            href="/reservation/schedule-usage/list"
            target="_self"
          >
            <!---->
            <!---->
             用途 
            <!---->
          </a>
        </li>
         
        <li
          class="c-sidebar-nav-item"
        >
          <a
            class="c-sidebar-nav-link"
            href="/reservation/schedule-timetable/list"
            target="_self"
          >
            <!---->
            <!---->
             時間割 
            <!---->
          </a>
        </li>
         
        <!---->
      </ul>
    </li>
    <div
      class="ps__rail-x"
      style="left: 0px; top: 0px;"
    >
      <div
        class="ps__thumb-x"
        style="left: 0px; width: 0px;"
        tabindex="0"
      />
    </div>
    <div
      class="ps__rail-y"
      style="top: 0px; left: 0px;"
    >
      <div
        class="ps__thumb-y"
        style="top: 0px; height: 0px;"
        tabindex="0"
      />
    </div>
  </ul>
</div>
`;

exports[`MySidebar 認証あり・権限あり reservation.write 表示 1`] = `
<div
  class="c-sidebar c-sidebar-dark c-sidebar-lg-show c-sidebar-fixed"
>
  <div
    class="c-sidebar-brand"
  >
    <span
      class="ml-2 font-weight-bold"
    >
      やそスポーツクラブ
    </span>
  </div>
   
  <ul
    class="c-sidebar-nav h-100 ps"
    data-v-3738b722=""
    style="position: relative;"
  >
    <li
      class="c-sidebar-nav-item"
      data-v-3738b722=""
    >
      <a
        aria-current="page"
        class="router-link-exact-active c-active c-sidebar-nav-link"
        href="/"
        target="_self"
      >
        <!---->
        <!---->
         ホーム 
        <!---->
      </a>
    </li>
     
    <li
      class="c-sidebar-nav-item"
      data-v-3738b722=""
    >
      <a
        class="c-sidebar-nav-link"
        href="/schedule/list"
        target="_self"
      >
        <!---->
        <!---->
         練習予定 
        <!---->
      </a>
    </li>
     
    <li
      class="c-sidebar-nav-dropdown"
      data-v-3738b722=""
    >
      <a
        class="c-sidebar-nav-dropdown-toggle"
      >
        <!---->
        <!---->
         会場予約 
      </a>
      <ul
        class="c-sidebar-nav-dropdown-items"
      >
        <li
          class="c-sidebar-nav-item"
        >
          <a
            class="c-sidebar-nav-link"
            href="/reservation/month/list"
            target="_self"
          >
            <!---->
            <!---->
             月一覧 
            <!---->
          </a>
        </li>
         
        <li
          class="c-sidebar-nav-item"
        >
          <a
            class="c-sidebar-nav-link"
            href="/reservation/schedule/list"
            target="_self"
          >
            <!---->
            <!---->
             練習予定一覧 
            <!---->
          </a>
        </li>
         
        <li
          class="c-sidebar-nav-item"
        >
          <a
            class="c-sidebar-nav-link"
            href="/reservation/reservation/list"
            target="_self"
          >
            <!---->
            <!---->
             会場予約一覧 
            <!---->
          </a>
        </li>
         
        <li
          class="c-sidebar-nav-item"
        >
          <a
            class="c-sidebar-nav-link"
            href="/reservation/schedule-status/list"
            target="_self"
          >
            <!---->
            <!---->
             公開ステータス 
            <!---->
          </a>
        </li>
         
        <li
          class="c-sidebar-nav-item"
        >
          <a
            class="c-sidebar-nav-link"
            href="/reservation/reservation-status/list"
            target="_self"
          >
            <!---->
            <!---->
             予約ステータス 
            <!---->
          </a>
        </li>
         
        <li
          class="c-sidebar-nav-item"
        >
          <a
            class="c-sidebar-nav-link"
            href="/reservation/reservation-organization/list"
            target="_self"
          >
            <!---->
            <!---->
             予約団体 
            <!---->
          </a>
        </li>
         
        <li
          class="c-sidebar-nav-item"
        >
          <a
            class="c-sidebar-nav-link"
            href="/reservation/schedule-place/list"
            target="_self"
          >
            <!---->
            <!---->
             場所 
            <!---->
          </a>
        </li>
         
        <li
          class="c-sidebar-nav-item"
        >
          <a
            class="c-sidebar-nav-link"
            href="/reservation/schedule-usage/list"
            target="_self"
          >
            <!---->
            <!---->
             用途 
            <!---->
          </a>
        </li>
         
        <li
          class="c-sidebar-nav-item"
        >
          <a
            class="c-sidebar-nav-link"
            href="/reservation/schedule-timetable/list"
            target="_self"
          >
            <!---->
            <!---->
             時間割 
            <!---->
          </a>
        </li>
         
        <li
          class="c-sidebar-nav-item"
        >
          <a
            class="c-sidebar-nav-link"
            href="/reservation/setting/edit"
            target="_self"
          >
            <!---->
            <!---->
             設定 
            <!---->
          </a>
        </li>
      </ul>
    </li>
    <div
      class="ps__rail-x"
      style="left: 0px; top: 0px;"
    >
      <div
        class="ps__thumb-x"
        style="left: 0px; width: 0px;"
        tabindex="0"
      />
    </div>
    <div
      class="ps__rail-y"
      style="top: 0px; left: 0px;"
    >
      <div
        class="ps__thumb-y"
        style="top: 0px; height: 0px;"
        tabindex="0"
      />
    </div>
  </ul>
</div>
`;

exports[`MySidebar 認証あり・権限なし 表示 1`] = `
<div
  class="c-sidebar c-sidebar-dark c-sidebar-lg-show c-sidebar-fixed"
>
  <div
    class="c-sidebar-brand"
  >
    <span
      class="ml-2 font-weight-bold"
    >
      やそスポーツクラブ
    </span>
  </div>
   
  <ul
    class="c-sidebar-nav h-100 ps"
    data-v-3738b722=""
    style="position: relative;"
  >
    <li
      class="c-sidebar-nav-item"
      data-v-3738b722=""
    >
      <a
        aria-current="page"
        class="router-link-exact-active c-active c-sidebar-nav-link"
        href="/"
        target="_self"
      >
        <!---->
        <!---->
         ホーム 
        <!---->
      </a>
    </li>
     
    <li
      class="c-sidebar-nav-item"
      data-v-3738b722=""
    >
      <a
        class="c-sidebar-nav-link"
        href="/schedule/list"
        target="_self"
      >
        <!---->
        <!---->
         練習予定 
        <!---->
      </a>
    </li>
     
    <!---->
    <div
      class="ps__rail-x"
      style="left: 0px; top: 0px;"
    >
      <div
        class="ps__thumb-x"
        style="left: 0px; width: 0px;"
        tabindex="0"
      />
    </div>
    <div
      class="ps__rail-y"
      style="top: 0px; left: 0px;"
    >
      <div
        class="ps__thumb-y"
        style="top: 0px; height: 0px;"
        tabindex="0"
      />
    </div>
  </ul>
</div>
`;

exports[`MySidebar 認証なし 表示 1`] = `
<div
  class="c-sidebar c-sidebar-dark c-sidebar-lg-show c-sidebar-fixed"
>
  <div
    class="c-sidebar-brand"
  >
    <span
      class="ml-2 font-weight-bold"
    >
      やそスポーツクラブ
    </span>
  </div>
   
  <ul
    class="c-sidebar-nav h-100 ps"
    data-v-3738b722=""
    style="position: relative;"
  >
    <li
      class="c-sidebar-nav-item"
      data-v-3738b722=""
    >
      <a
        aria-current="page"
        class="router-link-exact-active c-active c-sidebar-nav-link"
        href="/"
        target="_self"
      >
        <!---->
        <!---->
         ホーム 
        <!---->
      </a>
    </li>
     
    <li
      class="c-sidebar-nav-item"
      data-v-3738b722=""
    >
      <a
        class="c-sidebar-nav-link"
        href="/schedule/list"
        target="_self"
      >
        <!---->
        <!---->
         練習予定 
        <!---->
      </a>
    </li>
     
    <!---->
    <div
      class="ps__rail-x"
      style="left: 0px; top: 0px;"
    >
      <div
        class="ps__thumb-x"
        style="left: 0px; width: 0px;"
        tabindex="0"
      />
    </div>
    <div
      class="ps__rail-y"
      style="top: 0px; left: 0px;"
    >
      <div
        class="ps__thumb-y"
        style="top: 0px; height: 0px;"
        tabindex="0"
      />
    </div>
  </ul>
</div>
`;
